internal/affine/colorm_test.go

```diff
@@ -79,3 +80,94 @@ func TestColorMScaleOnly(t *testing.T) {
 		}
 	}
 }
+
+func TestColorMIsInvertible(t *testing.T) {
+	m := &ColorM{}
+	m = m.SetElement(1, 0, .5)
+	m = m.SetElement(1, 1, .5)
+	m = m.SetElement(1, 2, .5)
+	m = m.SetElement(1, 3, .5)
+	m = m.SetElement(1, 4, .5)
+	cidentity := &ColorM{}
+	//
+	cinvalid := &ColorM{}
+	cinvalid = cinvalid.SetElement(0, 0, 0)
+	cinvalid = cinvalid.SetElement(1, 1, 0)
+	cinvalid = cinvalid.SetElement(2, 2, 0)
+	cinvalid = cinvalid.SetElement(3, 3, 0)
+	//
+	cases := []struct {
+		In  *ColorM
+		Out bool
+	}{
+		{
+			nil,
+			true,
+		},
+		{
+			cidentity,
+			true,
+		},
+		{
+			m,
+			true,
+		},
+		{
+			cinvalid,
+			false,
+		},
+	}
+	for _, c := range cases {
+		got := c.In.IsInvertible()
+		want := c.Out
+		if got != want {
+			t.Errorf("%v.IsInvertible(): got: %t, want: %t", c.In, got, want)
+		}
+	}
+}
+
+func TestColorMInvert(t *testing.T) {
+	a := (&ColorM{}).SetElement(0, 0, 1).SetElement(1, 0, 8).SetElement(2, 0, -9).SetElement(3, 0, 7)
+	a = a.SetElement(0, 1, 0).SetElement(1, 1, 1).SetElement(2, 1, 0).SetElement(3, 1, 4)
+	a = a.SetElement(0, 2, 0).SetElement(1, 2, 0).SetElement(2, 2, 1).SetElement(3, 2, 2)
+	a = a.SetElement(0, 3, 0).SetElement(1, 3, 0).SetElement(2, 3, 0).SetElement(3, 3, 1)
+	a = a.SetElement(0, 4, 0).SetElement(1, 4, 0).SetElement(2, 4, 0).SetElement(3, 4, 0)
+
+	ia := (&ColorM{}).SetElement(0, 0, 1).SetElement(1, 0, -8).SetElement(2, 0, 9).SetElement(3, 0, 7)
+	ia = ia.SetElement(0, 1, 0).SetElement(1, 1, 1).SetElement(2, 1, 0).SetElement(3, 1, -4)
+	ia = ia.SetElement(0, 2, 0).SetElement(1, 2, 0).SetElement(2, 2, 1).SetElement(3, 2, -2)
+	ia = ia.SetElement(0, 3, 0).SetElement(1, 3, 0).SetElement(2, 3, 0).SetElement(3, 3, 1)
+	ia = ia.SetElement(0, 4, 0).SetElement(1, 4, 0).SetElement(2, 4, 0).SetElement(3, 4, 0)
+
```

Comment:
Defining elements by a slice and initializing ColorMs with them would be more readable.

Reply:
But the test package is `package affine_test` so it doesn't have access to the private `body` and `translate`. I would still have to call SetElement multiple times, right?

Reply:
Yes, so you'd need to call SetElement by for-loop.

```
es := [4][5]float32{ { ... }, { ... }, ... }
for j := 0; j < 5; j ++ {
  for i := 0; i < 4; i++ {
    ia = ia.SetElement(i, j, es[j][i])
  }
}
```